### ISSUE TITLE
Denom validation by cosmos SDK methods

### ISSUE THREAD

**Issue by Brando753** [Sifchain/sifnode#2901]
File Affected:  x/clp/types/asset.go: 23-34. Because the Amount field of sdk.Coin type is always sdk.OneInt(), building a coin type just to validate just the denom is redundant. Use sdk.ValidateDenom() and return a boolean based on any thrown error instead of using coin.Validate().

**Comment by Brando753** [comment]
This same ticket also affects the file: x/tokenregistry/types.msgs.go: 31-40, 80-89, 127-133.

**Comment by juniuszhou** [comment]
Just find out the MaxSymbolLength used in the clp module is different with cosmos sdk. So I suggest we won't fix it.

**Comment by pandaring2you** [comment]
Need 2 more confirmations from either @smartyalgo @Brando753 @banshee

**Comment by pandaring2you** [comment]
When we get 2 confirmations, will close the PR without merging.

**Comment by pandaring2you** [comment]
Note: @juniuszhou Should reach out to sifnode team why MaxSymbolLength is different than CosmosSDK

**Comment by banshee** [comment]
I'm wondering why we trim whitespace:

```
func (a Asset) Validate() bool {
	if !VerifyRange(len(strings.TrimSpace(a.Symbol)), 0, MaxSymbolLength) {
		return false
	}
	coin := sdk.NewCoin(a.Symbol, sdk.OneInt())
	return coin.IsValid()
}
```

This seems like a historical misfeature.  The original code returned false if the symbol just contained whitespace.  The new code checks the length of the symbol after removing whitespace, so a symbol thats "<1000 spaces>SYM<1000 spaces>" is valid.  Seems weird.

**Comment by banshee** [comment]
Believe @Brando753 meant x/tokenregistry/types/msgs.go, where we construct coins just to validate denom.

**Comment by banshee** [comment]
Also see https://github.com/Sifchain/sifnode/pull/3085

**Comment by juniuszhou** [comment]
wait the input from sifnode team if MaxSymbolLength=71 is reasonable and should we remove it.

**Comment by juniuszhou** [comment]
@timlind and @AustinoBombino agree on extend it. #3028 ready for code review.

**Comment by banshee** [comment]
Merged #3028

**Comment by juniuszhou** [comment]
close it after PR #3028 merged.
